src/bootstrap/src/core/builder.rs

```diff
@@ -438,8 +438,27 @@ impl StepDescription {
             }
         }
 
-        // strip CurDir prefix if present
-        let mut paths: Vec<_> = paths.iter().map(|p| p.strip_prefix(".").unwrap_or(p)).collect();
+        // Resolve paths to be relative to the builder source directory.
+        let paths: Vec<_> = paths
+            .iter()
+            .map(|p| {
+                // If the path does not exist, it may represent the name of a Step, such as `tidy` in `x test tidy`
+                if !p.exists() {
+                    return p.clone();
+                }
+                // Get the canonical path, strip the prefix, and convert to a PathBuf.
+                match p.canonicalize() {
+                    Ok(p) => p.strip_prefix(&builder.src).unwrap_or(&p).to_owned(),
+                    Err(e) => {
+                        eprintln!("ERROR: {:?}", e);
+                        panic!("Due to the above error, failed to canonicalize path: {:?}", p);
```

Comment:
@bjorn3 I am still using `.canonicalize` here.

I think this issue describes the problem you were referring to: https://github.com/rust-lang/rust/issues/42869

Some comments on that issue recommend using `std::path::absolute`, but my IDE tells me this is an unstable library feature.

```suggestion
                // Get the absolute path, strip the prefix, and convert to a PathBuf.
                match std::path::absolute(p) {
                    Ok(p) => p.strip_prefix(&builder.src).unwrap_or(&p).to_owned(),
                    Err(e) => {
                        eprintln!("ERROR: {:?}", e);
                        panic!("Due to the above error, failed to resolve path: {:?}", p);
```

Reply:
It was stabilized in 1.79.

Reply:
That's great news! I just pushed some changes :D